XRTK-Core/Assets/XRTK.Tests/Core/TestFixture_02_MixedRealityToolkitUtilityTests.cs

```diff
@@ -0,0 +1,70 @@
+
+using NUnit.Framework;
+using System;
+using XRTK.Definitions.InputSystem;
+using XRTK.Definitions.Utilities;
+using XRTK.Extensions;
+using XRTK.Providers.Controllers.OpenVR;
+using XRTK.Services;
+using XRTK.Tests.Services;
+using XRTK.Utilities;
+
+namespace XRTK.Tests.Core
+{
+    public class TestFixture_02_MixedRealityToolkitUtilityTests
+    {
+        #region Configuration Validation Tests
+
+        [Test]
+        public void Test_01_ConfirmDataProviderConfigurationNotPresent()
+        {
+            Type[] dataProviderTypes = new[] { typeof(TestDataProvider1) };
+            ControllerDataProviderConfiguration[] dataProviderConfiguration = new[] { new ControllerDataProviderConfiguration(typeof(TestDataProvider1), "Test Controller Data Provider", 2, SupportedPlatforms.WindowsStandalone | SupportedPlatforms.Editor, null) };
+
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            InitializeDefaultInputSystemProfile();
+
+            Assert.IsFalse(ValidateConfiguration.ValidateDataProviders(dataProviderTypes, dataProviderConfiguration, false));
+        }
+
+        [Test]
+        public void Test_02_ConfirmDataProviderConfigurationPresent()
+        {
+            Type[] dataProviderTypes = new[] { typeof(TestDataProvider1) };
+            ControllerDataProviderConfiguration[] dataProviderConfiguration = new[] { new ControllerDataProviderConfiguration(typeof(TestDataProvider1), "Test Controller Data Provider", 2, SupportedPlatforms.WindowsStandalone | SupportedPlatforms.Editor, null) };
+
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            InitializeDefaultInputSystemProfile();
+
+            MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile.ControllerDataProvidersProfile.RegisteredControllerDataProviders = MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile.ControllerDataProvidersProfile.RegisteredControllerDataProviders.AddItem<ControllerDataProviderConfiguration>(dataProviderConfiguration[0]);
+
+            Assert.IsTrue(ValidateConfiguration.ValidateDataProviders(dataProviderTypes, dataProviderConfiguration, false));
+        }
+
+        [Test]
+        public void Test_03_ConfirmControllerMappingConfigurationNotPresent()
+        {
+            Type[] controllerTypes = new[] { typeof(GenericOpenVRController) };
+            ControllerDataProviderConfiguration[] dataProviderConfiguration = new[] { new ControllerDataProviderConfiguration(typeof(TestDataProvider1), "Test Controller Data Provider", 2, SupportedPlatforms.WindowsStandalone | SupportedPlatforms.Editor, null) };
+
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            InitializeDefaultInputSystemProfile();
+
+            Assert.IsFalse(ValidateConfiguration.ValidateControllerProfiles(controllerTypes, false));
+        }
+
+        private void InitializeDefaultInputSystemProfile()
+        {
+            var inputSystemProfile = new MixedRealityInputSystemProfile();
```

Comment:
You can't use UnityEngine types like this

Reply:
should have been:
```
var inputSystemProfile = ScriptableObject.CreateInstance<MixedRealityInputSystemProfile>();
            inputSystemProfile.ControllerDataProvidersProfile = ScriptableObject.CreateInstance<MixedRealityControllerDataProvidersProfile>();
            inputSystemProfile.ControllerMappingProfiles = ScriptableObject.CreateInstance<MixedRealityControllerMappingProfiles>();
            ```

Reply:
https://github.com/JetBrains/resharper-unity/wiki/ScriptableObjects-must-be-instantiated-with-ScriptableObject.CreateInstance-instead-of-new

Reply:
This will matter when we start doing runtime unit tests